src/app/app.py

```diff
@@ -10,38 +10,38 @@
 from src.app.archbishop import archbishop
 
 
-# Initialise app
-app = Flask(__name__)
+def create_app():
+    # Initialise app
+    app = Flask(__name__)
 
+    # Initialise config
+    config = Config()
+    app.config.from_object(config)
+    app.app_context().push()
 
-# Initialise config
-config = Config()
-app.config.from_object(config)
-app.app_context().push()
+    # Initialise slack
+    slack_client = WebClient(token=config.BOT_TOKEN)
+    slack_events_adapter = SlackEventAdapter(config.SECRET, config.EVENT_PATH, app)
 
+    # Initialise redis
+    redis = RedisClient(app, config.REDIS_URL, config.REDIS_TOKEN)
 
-# Initialise slack
-slack_client = WebClient(token=config.BOT_TOKEN)
-slack_events_adapter = SlackEventAdapter(config.SECRET, config.EVENT_PATH, app)
+    # Initialise logger
+    dictConfig(json.load(open(from_root('src', 'app', 'config', 'logs.json'))))
+    app.logger_name = 'waffle-bot'
 
+    # Initialise messages
+    messages = Messages(from_root('src', 'app', 'resources', 'messages.properties'))
 
-# Initialise redis
-redis = RedisClient(app, config.REDIS_URL, config.REDIS_TOKEN)
+    # Register blueprint
+    app.register_blueprint(archbishop.construct_blueprint(slack_events_adapter, config, messages, redis))
 
+    return app
 
-# Initialise logger
-dictConfig(json.load(open(from_root('app', 'config', 'logs.json'))))
-app.logger_name = 'waffle-bot'
 
+# Create the Flask app
+app = create_app()
 
-# Initialise messages
-messages = Messages(from_root('resources', 'messages.properties'))
-
-
-# Register blueprint
-app.register_blueprint(archbishop.construct_blueprint(slack_events_adapter, config, messages, redis))
-
-
-# Let's go!
+# Only get to here when running directly
```

Comment:
So when will this get used?

Reply:
I think only if you can get it started on your host